projects/schematics/src/migrations/2_0/ssr/ssr_spec.ts

```diff
@@ -0,0 +1,211 @@
+    appSchematicRunner = new SchematicTestRunner(
+      'test',
+      require.resolve('../../migrations.json')
+    );
+
+    appTree = await appSchematicRunner
+      .runExternalSchematicAsync(
+        '@schematics/angular',
+        'workspace',
+        workspaceOptions
+      )
+      .toPromise();
+    appTree = await appSchematicRunner
+      .runExternalSchematicAsync(
+        '@schematics/angular',
+        'application',
+        appOptions,
+        appTree
+      )
+      .toPromise();
+
+    appTree.create('/server.ts', 'server content');
+    appTree.create('/webpack.server.config.js', 'webpack config content');
+
+    appTree.create(
+      '/src/main.server.ts',
+      `
+    import { enableProdMode } from '@angular/core';
+    import { environment } from './environments/environment';
+    if (environment.production) {
+      enableProdMode();
+    }
+    export { AppServerModule } from './app/app.server.module';
+    export { renderModule, renderModuleFactory } from '@angular/platform-server';
+    export { ngExpressEngine } from '@nguniversal/express-engine';
+    export { provideModuleMap } from "@nguniversal/module-map-ngfactory-loader";
+    `
+    );
+
+    appTree.create(
+      '/src/app/app.server.module.ts',
+      `
+    import { NgModule } from '@angular/core';
+    import { ServerModule } from '@angular/platform-server';
+    import { AppModule } from './app.module';
+    import { AppComponent } from './app.component';
+    import { ModuleMapLoaderModule } from '@nguniversal/module-map-ngfactory-loader';
+    @NgModule({
+      imports: [
+        AppModule,
+        ServerModule,
+        ModuleMapLoaderModule,
+      ],
+      bootstrap: [AppComponent],
+    })
+    export class AppServerModule {}
+    `
+    );
+
+    const pkg = JSON.parse(appTree.readContent('/package.json'));
+    const scripts = pkg.scripts;
+    scripts['compile:server'] = 'old compile:server';
+    scripts['serve:ssr'] = 'old serve:ssr';
+    scripts['build:client-and-server-bundles'] =
+      'old build:client-and-server-bundles';
+
+    appTree.overwrite('/package.json', JSON.stringify(pkg, null, 2));
+
+    const angularJson = JSON.parse(appTree.readContent('/angular.json'));
+    angularJson.projects[appOptions.name].architect['server'] = {
+      options: {
+        outputPath: `dist/${appOptions.name}/server`,
+      },
+    };
+
+    appTree.overwrite('angular.json', JSON.stringify(angularJson, null, 2));
+  });
+
+  describe('Schematics SSR migration', () => {
+    it(`shouldn't perform migration if server-side is not available`, async () => {
+      await appTree.delete('/server.ts');
+      const angularJson = JSON.parse(appTree.readContent('/angular.json'));
+      delete angularJson.projects[appOptions.name].architect['server'];
+      await appTree.overwrite(
+        'angular.json',
+        JSON.stringify(angularJson, null, 2)
+      );
+
+      await runMigration(appTree, appSchematicRunner, 'migration-v2-ssr-07');
```

Comment:
We can extract this `migration-v2-ssr-07` to a const as well

Reply:
Done 👍